app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/BankAccountController.scala

```diff
@@ -81,29 +80,51 @@ class BankAccountController @Inject() (
 
   def checkBankAccountDetails(implicit journey: JourneyBindable): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
-      withAnswersAndRoutes[DisplayDeclaration] { (_, declaration, router) =>
-        val answers = declaration.flatMap(p => p.displayResponseDetail.maskedBankDetails)
-        answers.fold(Redirect(router.nextPageForCheckBankAccountDetails())) { maskedBankDetails =>
-          (maskedBankDetails.consigneeBankDetails, maskedBankDetails.declarantBankDetails) match {
-            case (Some(cmbd), _)    =>
-              Ok(
-                checkBankAccountDetailsPage(
-                  BankAccountAcc14(cmbd.accountHolderName, cmbd.sortCode, cmbd.accountNumber),
-                  continuePage.url,
-                  router
+      withAnswersAndRoutes[DisplayDeclaration] { (fillingOutClaim, declaration, router) =>
+        val isChange = fillingOutClaim.draftClaim.fold(identity).bankAccountDetailsAnswer.nonEmpty
+        isChange match {
+          case true  =>
+            fillingOutClaim.draftClaim.fold(identity).bankAccountDetailsAnswer match {
+              case Some(bankAccountDetails: BankAccountDetails) =>
+                Ok(
+                  checkBankAccountDetailsPage(
+                    BankAccountView(
+                      bankAccountDetails.accountName.value,
+                      bankAccountDetails.sortCode.value,
+                      bankAccountDetails.accountNumber.value
+                    ),
+                    continuePage.url,
+                    router
+                  )
                 )
-              )
-            case (None, Some(dmbd)) =>
-              Ok(
-                checkBankAccountDetailsPage(
-                  BankAccountAcc14(dmbd.accountHolderName, dmbd.sortCode, dmbd.accountNumber),
-                  continuePage.url,
-                  router
-                )
-              )
-            case (None, None)       =>
-              Redirect(router.nextPageForCheckBankAccountDetails())
-          }
+              case None                                         => Redirect(router.nextPageForCheckBankAccountDetails())
+            }
+          case false =>
+            val answers = declaration.flatMap(p => p.displayResponseDetail.maskedBankDetails)
+
+            answers.fold(Redirect(router.nextPageForCheckBankAccountDetails())) { maskedBankDetails =>
+              (maskedBankDetails.consigneeBankDetails, maskedBankDetails.declarantBankDetails) match {
+                case (Some(cmbd), _)    =>
+                  Ok(
+                    checkBankAccountDetailsPage(
+                      BankAccountView(cmbd.accountHolderName, cmbd.sortCode, cmbd.accountNumber),
+                      continuePage.url,
+                      router
+                    )
+                  )
+                case (None, Some(dmbd)) =>
```

Comment:
This case looks exactly like the one above. Can you collapse them?
Maybe:
`case (Some(x),None) | (None,Some(x) =>`

Reply:
In fact they are different as they are pulling different bank account details depending on whether there are consignee details or declarant details